python/ray/_private/worker.py

```diff
@@ -2309,7 +2311,9 @@ def connect(
     elif not LOCAL_MODE:
         raise ValueError("Invalid worker mode. Expected DRIVER, WORKER or LOCAL.")
 
-    gcs_options = ray._raylet.GcsClientOptions.from_gcs_address(node.gcs_address)
+    gcs_options = ray._raylet.GcsClientOptions.create(
+        node.gcs_address, node._ray_params.cluster_id
```

Comment:
This should be `node.cluser_id` which is guaranteed to be not Nil but `node._ray_params.cluster_id` can be Nil?

Also add a check to make sure it's not Nil.

Reply:
after some test: this really can be Nil, if the worker is a driver worker and gcs is created from ray.init()

Reply:
changed to node.cluster_id, waiting ci